pkg/controller/csi_handler.go

```diff
@@ -458,9 +482,18 @@ func (h *csiHandler) getNodeID(driver string, nodeName string) (string, error) {
 
 	// Check Node annotation.
 	node, err := h.nodeLister.Get(nodeName)
-	if err != nil {
+	if err == nil {
+		return GetNodeIDFromNode(driver, node)
+	}
+
+	// Check VolumeAttachment annotation as the last resort
+	if va == nil {
 		return "", err
 	}
+	if nodeID, found := va.Annotations[vaNodeIDAnnotation]; found {
+		return nodeID, nil
+	}
 
-	return GetNodeIDFromNode(driver, node)
+	// return nodeLister.Get error
+	return "", err
```

Comment:
same comment as above, maybe annotate with fact VA also not found

Reply:
same answer as above :-)